src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -531,6 +537,37 @@ public void delete(String modelId, ActionListener<DeleteModelResponse> listener)
                     return;
                 }
 
+                // Get index mappings to check if model is being used by any indices
+                getMappings(modelId, getMappingsStep);
+            }, listener::onFailure);
+
+            getMappingsStep.whenComplete(getMappingsResponse -> {
+                // The mapping metadata string will always have the model id followed by `,` and another parameter, or the end of the object
```

Comment:
An alternative solution would be to parse the map through casting and null checks, then check if the model_id field exists and then if it's equal to the modelId being deleted.  I chose to go with contains, since there are only two possible characters that can follow the id since the mapping string is in json format.  The other reason is that casting could get complicated with multiple fields beneath properties and would end up with a lot of nested loops.

Reply:
can you try to use regex with a capturing group here something like `.*modelId=(.*)(?=\s|\t|,|})` and then reference the capturing group to get the id itself?  That would be less code than tokenizing everything

Regex will be safer here if there is a space or tab in between on accident considering its a json

Reply:
That's true, I'm going to attempt the casting solution first, but will use this if that is not feasible.